test/integration/basic_test.go

```diff
@@ -99,7 +106,13 @@ func initMockUP4() (err error) {
 	return nil
 }
 
-func setup(t *testing.T) {
+func setup(t *testing.T, pfcpAgentConfig string) {
+	providers.RunDockerCommandAttach("pfcpiface",
+		fmt.Sprintf("/bin/pfcpiface -config /config/%s", pfcpAgentConfig))
+
+	// wait for PFCP Agent to initialize
+	time.Sleep(time.Second * 3)
```

Comment:
This might not be feasible, but doing a real ready check would be faster and less flaky than a simple sleep. Now that `setup` is run on every test case, these 3 seconds add up quickly.

On http servers people do a simple port check with netcat.